source/customization.rst

```diff
@@ -966,3 +966,54 @@ Changes are applied by running ``update_ood_portal`` and restarting the ``ondema
    sudo systemctl restart ondemand-dex.service
 
 .. _dex template docs: https://github.com/dexidp/dex/blob/master/Documentation/templates.md
+
+.. _xdmod_integration:
+
+XDMoD Integration (BETA)
+------------------------
+
+XDMoD Integration requires XDMoD 9+, OnDemand 1.8+, and the ability to facilitate single sign on between the two services. Currently this has been demonstrated to work using OpenID Connect via Keycloak as well as a modified instance of Dex Identity Provider to support sessions.
+
+.. figure:: /images/customization_xdmod.png
+   :align: center
+
+   Example of XDMoD Job Efficiency reports in the OnDemand Dashboard.
+
+Steps to enable the XDMoD reports in the OnDemand Dashboard:
+
+#. Configure OnDemand with XDMoD host URL in PUN /etc/ood/config/nginx_stage.yml
+
+   .. code-block:: yaml
+
+      pun_custom_env:
+        OOD_XDMOD_HOST: "https://xdmod.osc.edu"
+
+#. Add OnDemand host as domain to XDMoD portal settings for CORS /etc/xdmod/portal_settings.ini
+
+   .. code-block:: none
+
+      domains = "https://ondemand.osc.edu"
+
+#. Configure identity provider to include OnDemand host in HTTP Content-Security-Policy for frame-ancestors since OnDemand uses iFrames to trigger SSO with XDMoD when a user logs in
+
+   .. code-block:: none
+
+      frame-ancestors https://*.osc.edu 'self'
+
```

Comment:
Is this code example for KeyCloak or Dex?

Reply:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/frame-ancestors

It is the value of the Content-Security-Policy header

It was what we had to set in Keycloak but I guess the full header would look like

    Content-Security-Policy: frame-ancestors https://*.osc.edu 'self'

Reply:
I added a link to MDN and clarified the example was the value of the Content-Security-Policy header that was used in Keycloak at OSC